solutions/community-deploy/cloud/AWS/README.md

```diff
@@ -55,15 +63,15 @@ tag_name            = "cortx-multinode"
 ```
 terraform validate && terraform apply -var-file user.tfvars --auto-approve
 ```
-- Execute the following commands to find the Public and Private ip addresses from local host by setting as environment variable.
-
-**Public ip address:**
-```
-PUBLIC_IP=`terraform show -json terraform.tfstate | jq .values.outputs.aws_instance_public_ip_addr.value 2>&1 | tee ip_public.txt | tr -d '",[]' | sed '/^$/d'`
+- Execute the following commands as environment variables which will be used later the deployment
 ```
-**Private ip address:**
-```
-PRIVATE_IP=`terraform show -json terraform.tfstate | jq .values.outputs.aws_instance_private_ip_addr.value 2>&1 | tee ip_private.txt | tr -d '",[]' | sed '/^$/d'`
+export PUBLIC_IP=$(terraform show -json terraform.tfstate | jq .values.outputs.aws_instance_public_ip_addr.value 2>&1 | tee ip_public.txt | tr -d '",[]' | sed '/^$/d')
+export PRIMARY_PUBLIC_IP=$(cat ip_public.txt | jq '.[0]'| tr -d '",[]')
+export WORKER_IP=$(cat ip_public.txt | jq '.[0]','.[1]','.[2]','.[3]' | tr -d '",[]')
+export HOST1=$(cat ec2_hostname.txt | jq '.[0]'| tr -d '",[]')
```

Comment:
This will work only for the `1+3` setup.  This will fail if the developer uses more nodes. Hence we should use a loop here.

Reply:
fixed

Reply:
It still shows hardcoded variable declaration for `HOST1`, `HOST2` etc. 

```
export HOST1=$(cat ec2_hostname.txt | jq '.[0]'| tr -d '",[]')
export HOST2=$(cat ec2_hostname.txt | jq '.[1]'| tr -d '",[]')
export HOST3=$(cat ec2_hostname.txt | jq '.[2]'| tr -d '",[]')
export HOST4=$(cat ec2_hostname.txt | jq '.[3]'| tr -d '",[]')
```

Following command in https://github.com/mukul-seagate11/cortx-re-1/tree/mukul-communitybuild-steps/solutions/community-deploy/cloud/AWS#cortx-deployment uses above varibales 

```
export ROOT_PASSWORD=<YOUR_ROOT_PASSWORD>
for ip in $PUBLIC_IP;do ssh -i cortx.pem -o 'StrictHostKeyChecking=no' centos@"${ip}" "git clone https://github.com/Seagate/cortx-re && pushd /home/centos/cortx-re/solutions/kubernetes && touch hosts && echo hostname=${HOST1},user=root,pass= > hosts && sed -i 's,pass=.*,pass=$ROOT_PASSWORD,g' hosts && echo hostname=${HOST2},user=root,pass= >> hosts && sed -i 's,pass=.*,pass=$ROOT_PASSWORD,g' hosts && echo hostname=${HOST3},user=root,pass= >> hosts && sed -i 's,pass=.*,pass=$ROOT_PASSWORD,g' hosts && echo hostname=${HOST4},user=root,pass= >> hosts && sed -i 's,pass=.*,pass=$ROOT_PASSWORD,g' hosts && cat hosts";done
```

Reply:
fixed

Reply:
@mukul-seagate11 Looks good now.  However, I feel we need to use PRIVATE_IP for updating `hosts` file to be used for K8s and CORTX deployment. 
in https://github.com/mukul-seagate11/cortx-re-1/tree/mukul-communitybuild-steps/solutions/community-deploy/cloud/AWS#cortx-deployment

Reply:
It should not be PRIVATE_IP as it will not connect from local so connecting through PUBLIC _IP & then private hostname will be updated

Reply:
so, no change required